From ed0af36f78f0cc9b481966e162247cf466230486 Mon Sep 17 00:00:00 2001
From: Emil Tantilov <emil.s.tantilov@intel.com>
Date: Fri, 18 Jan 2013 02:16:41 +0000
Subject: [PATCH 780/858] ixgbe: fix possible data corruption in read_i2c_byte

This patch makes sure that the SW lock is released after all i2c
operations complete in the retry code path.

Signed-off-by: Emil Tantilov <emil.s.tantilov@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit d0310dc415a670ae2b2e27076b4685d988830479)

Signed-off-by: Arnaldo Carvalho de Melo <acme@redhat.com>
---
 drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c b/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
index eb534a0..d7b5bd0 100644
--- a/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
+++ b/drivers/net/ethernet/intel/ixgbe/ixgbe_phy.c
@@ -1307,9 +1307,9 @@ s32 ixgbe_read_i2c_byte_generic(struct ixgbe_hw *hw, u8 byte_offset,
   break;
 
 fail:
+  ixgbe_i2c_bus_clear(hw);
   hw->mac.ops.release_swfw_sync(hw, swfw_mask);
   msleep(100);
-  ixgbe_i2c_bus_clear(hw);
   retry++;
   if (retry < max_retry)
    hw_dbg(hw, "I2C byte read error - Retrying.\n");
-- 
1.8.3.1